pallenec

```diff
@@ -63,15 +64,42 @@ local inspect_opts = {
     end
 }
 
+local function print_tokens(lexer)
+    while true do
+        local token = lexer:next()
+        if token.name == "EOF" then
+            break
+        end
+
+        local loc = token.loc
+        local end_loc = token.end_loc
+        local value = ""
+        if token.value then
+            local s = token.value
+            if type(token.value) == "string" then
+                s = C.string(token.value)
+            end
+            value = string.format(" %s", s)
+        end
+        print(string.format("%d:%d,%d:%d,%d:%d %s%s", loc.line, loc.col,
+            end_loc.line, end_loc.col, loc.pos, end_loc.pos, token.name, value))
+    end
+end
+
 local function dump(filename, stop_after)
     local input, err = driver.load_input(filename)
     if err then util.abort(err) end
 
-    local out, errs = driver.compile_internal(filename, input, stop_after, opt_level)
-    if not out then util.abort(table.concat(errs, "\n")) end
+    if stop_after == "lexer" then
+        local lexer = driver.compile_internal(filename, input, stop_after, opt_level)
+        print_tokens(lexer)
+    else
+        local out, errs = driver.compile_internal(filename, input, stop_after, opt_level)
```

Comment:
The local `out, errs` is the same in the two branches of the `if`. We could put it before the `if`.